### ISSUE TITLE
Documentation of featureFilter looks wrong

### ISSUE THREAD

**Issue by enersis-pst** [mapbox/mapbox-gl-js#12255]
have a look at the page of featureFilter
https://github.com/mapbox/mapbox-gl-js/tree/main/src/style-spec/feature_filter

it tolds to run `return testFilter({zoom: 0}, feature);`
for me this only works by run `return testFilter.filter({zoom: 0}, feature);`

also this filter doesnt work for me when using `geometry-type`:
```
[
    "in",
    [
        "geometry-type"
    ],
    [
        "literal",
        [
            "MultiLineString",
            "LineString"
        ]
    ]
]
```

by this feature
```
{
    "id": "1",
    "type": "Feature"
    "properties": {},
    "geometry": {
        "type": "LineString",
        "coordinates": [
            [
                8.859324306249619,
                47.84804269232009
            ],
            [
                8.864152282476425,
                47.8486042981051
            ],
            [
                8.864602893590927,
                47.85067786678525
            ],
            [
                8.857736438512802,
                47.85129704135821
            ],
            [
                8.859324306249619,
                47.84804269232009
            ]
        ]
    }
}
```
Looks like ive to convert the feature type from `Feature` to a number: 
```
const geometryTypes = [
    'Unknown', //0
    'Point', //1
    'LineString', //2
    'Polygon' //3
];
```

so, this works, but looks strange to me. 
```
{
    "id": "1",
    "type": 2 //<=========
    "properties": {},
    "geometry": {
        "type": "LineString",
        "coordinates": [...]
    }
}
```

Should it be like this?

**Comment by enersis-pst** [comment]
ok i think the feature here is not the same as like geojson Feature.
Would be nice to have this info in the docu.


https://github.com/mapbox/mapbox-gl-js/blob/main/src/style-spec/expression/index.js
```
export type Feature = {
    +type: 1 | 2 | 3 | 'Unknown' | 'Point' | 'MultiPoint' | 'LineString' | 'MultiLineString' | 'Polygon' | 'MultiPolygon',
    +id?: number | null,
    +properties: {[_: string]: any},
    +patterns?: {[_: string]: {"min": string, "mid": string, "max": string}},
    +geometry?: Array<Array<Point>>
};
```

**Comment by SnailBones** [comment]
@enersis-pst Thank you for bringing our attention to these issues! Both should be fixed in #12257, let me know if that helps.

**Comment by enersis-pst** [comment]
@SnailBones thanks for quick solving.
also docu change for featureFilter would be nice to know the exact required feature interface.
